yt/yt/ytlib/api/native/client_job_info_impl.cpp

```diff
@@ -1461,6 +1480,10 @@ static std::vector<TJob> ParseJobsFromArchiveResponse(
             job.Address = record.Address.value_or("");
         }
 
+        if (responseIdMapping.Addresses && record.Addresses) {
```

Comment:
This condition should be just `record.Addresses`. We plan to drop `This field previously was non-optional` compats later for other fields as well.